web/src/features/hfiCalculator/pages/HfiCalculatorPage.tsx

```diff
@@ -255,6 +262,23 @@ const HfiCalculatorPage: React.FunctionComponent = () => {
     }
   }
 
+  const buildNonReadyState = () => {
+    if (isUndefined(result) && isUndefined(selectedFireCentre)) {
```

Comment:
Already doing an isUndefined check down below.
What about a flow where it's:
```
if (isUndefined(selectedFireCentre)) {
   return gimmeTheEmptyFireCentreRow()
}
else if (loading || stationDataLoading || isUndefined(result)) {
   return gimmeTheLoadingCircle()
}
else {
  return gimmeTheTableWithAllTheStuff()
}
```

Reply:
Sonarcloud will complain about complexity in the component JSX :(